File: app/models/enumeration/IssueFilterType.java
package models.enumeration;

public enum IssueFilterType {
    ASSIGNED("assigned"),
    CREATED("created"),
    MENTIONED("mentioned"),
    FAVORITE("favorite"),
    ALL("all");

    private String issueFilter;

    IssueFilterType(String issueFilter) {
        this.issueFilter = issueFilter;
    }

    public static IssueFilterType getValue(String value) {
        for (IssueFilterType issueFilterType : IssueFilterType.values()) {
            if (issueFilterType.issueFilter.equals(value)) {
                return issueFilterType;
            }
        }
        throw new IllegalArgumentException("No matching issue filter type found for [" + value + "]");
    }
}

File: src/main/kotlin/com/github/yonaprojects/yona/domain/issue/IssueFilterType.kt
package com.github.yonaprojects.yona.domain.issue

// yona models/enumeration/IssueFilterType.java 대응 (P2-52) — UserApi.getIssuesByUser()의
// filter 쿼리 파라미터 값(assigned/created/mentioned/favorite/all)과 1:1 매핑된다.
enum class IssueFilterType(val filterValue: String) {
    ASSIGNED("assigned"),
    CREATED("created"),
    MENTIONED("mentioned"),
    FAVORITE("favorite"),
    ALL("all");

    companion object {
        fun getValue(value: String): IssueFilterType =
            entries.find { it.filterValue == value }
                ?: throw IllegalArgumentException("No matching issue filter type found for [$value]")
    }
}
